File: src/main/java/com/linebot/repository/BotUserRepository.java
package com.linebot.repository;

import com.linebot.entity.BotUser;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;


@Repository
public interface BotUserRepository extends JpaRepository<BotUser, Long> {
    List<BotUser> findByDeleted(String deleted);
    BotUser findByUserId(String userId);
}

File: src/main/kotlin/com/linebot/repository/BotUserRepository.kt
package com.linebot.repository

import com.linebot.entity.BotUser
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface BotUserRepository : JpaRepository<BotUser, Long> {
    fun findByDeleted(deleted: String): List<BotUser>
    fun findByUserId(userId: String): BotUser?
    fun findByAppUserIdAndDeleted(appUserId: String, deleted: String): BotUser?
}
